app/src/main/java/com/trueedu/project/di/RemoteModule.kt

```diff
@@ -0,0 +1,107 @@
+import com.trueedu.project.repository.remote.service.AuthService
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import kotlinx.serialization.json.Json
+import okhttp3.Headers
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import javax.inject.Singleton
+
+@InstallIn(SingletonComponent::class)
+@Module
+object RemoteModule {
+    @Provides
+    @Singleton
+    @NormalService
+    fun providesAuthService(retrofit: Retrofit): AuthService {
+        return retrofit.create(AuthService::class.java)
+    }
+
+    @Provides
+    @Singleton
+    @TokenRefreshService
+    fun providesTokenRefreshService(
+        @BaseUrl baseUrl: String,
+        httpLoggingInterceptor: HttpLoggingInterceptor,
+        chuckerInterceptor: ChuckerInterceptor,
+        @AppVersion appVersion: String?,
+        @AppVersionCode appVersionCode: Long
+    ): AuthService {
+        val contentType = "application/json".toMediaType()
+        val json = Json {
+            ignoreUnknownKeys = true
+            prettyPrint = true
+            isLenient = false
+        }
+
+        return Retrofit.Builder()
+            .baseUrl(baseUrl)
+            .client(
+                OkHttpClient.Builder()
+                    .addInterceptor(httpLoggingInterceptor)
+                    .addInterceptor(chuckerInterceptor)
+                    .addInterceptor { chain ->
+                        chain.request().newBuilder()
+                            //.headers(getApiHeaders(appVersion = appVersion, appVersionCode = appVersionCode))
+                            .build()
+                            .let {
+                                chain.proceed(it)
+                            }
+                    }
+                    .build()
+            )
+            .addConverterFactory(json.asConverterFactory(contentType))
+            .build()
+            .create(AuthService::class.java)
+    }
+
+    @Singleton
+    @Provides
+    fun providesAuthRemote(
+        @NormalService
+        authService: AuthService
+    ): AuthRemote = AuthRemoteImpl(authService = authService)
+}
+
+fun getApiHeaders(accessToken: String? = null, appVersion: String?, appVersionCode: Long): Headers {
+    val headers = Headers.Builder()
+
+    accessToken?.takeIf {
+        checkHeaderValid(it) && it.isNotBlank()
+    }?.let {
+        headers.set("Authorization", "Bearer $it")
+    }
+
+    return headers
+        .set(
+            "X-USERAGENT",
+            "$appVersion" +
+                    "/$appVersionCode Android" +
+                    "/${Build.VERSION.SDK_INT}" +
+                    "/${Build.MANUFACTURER}-${Build.MODEL.trim().replace(" ", "")}"
+        )
+        .build()
+}
+
+private fun checkHeaderValid(value: String): Boolean {
+    var result = true
+    for (i in value.indices) {
+        val c = value[i]
```

Comment:
https://kotlinandroid.org/kotlin/kotlin-iterate-over-characters-in-string/ 에 따르면 문자열 순회하면 char 하나씩 순회가능한듯합니다.

```suggestion
    for (c in value) {
```